colab-notebooks-examples/integrations/openai/intro.ipynb

```diff
@@ -0,0 +1,440 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f96b815e",
+   "metadata": {},
+   "source": [
+    "# Integrating Elasticsearch with OpenAI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e0f537af",
+   "metadata": {},
+   "source": [
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/elastic/elasticsearch-labs/blob/main/colab-notebooks-examples/integrations/openai/intro.ipynb)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "349e0e74",
+   "metadata": {},
+   "source": [
+    "This notebook demonstrates how to: \n",
+    "- Index the OpenAI Wikipedia vector dataset into Elasticsearch \n",
+    "- Embed a question with the OpenAI [`embeddings`](https://platform.openai.com/docs/api-reference/embeddings) endpoint\n",
+    "- Perform [kNN search](https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html) on the Elasticsearch index using the encoded question."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "690975de",
+   "metadata": {},
+   "source": [
+    "# Create Elastic Cloud deployment\n",
+    "If you don't have an Elastic Cloud deployment, sign up [here](https://cloud.elastic.co/registration?fromURI=%2Fhome) for a free trial.\n",
+    "\n",
+    "* Go to the [Create deployment](https://cloud.elastic.co/deployments/create) page\n",
+    "  * Select Create deployment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de32a789",
+   "metadata": {},
+   "source": [
+    "# Connect to Elasticsearch\n",
+    "\n",
+    "To get started with Elasticsearch, we will need to connect to Elastic deployment using the [python client](https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/index.html). Since we are using an Elastic Cloud deployment, we will use the Cloud ID to identify our deployment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cb38a868",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python3 -m pip install -q Elasticsearch"
```

Comment:
install all dependencies all at once. You can do multiple package installs with one command 

Example:
```
!pip install -qU elasticsearch sentence-transformers==2.2.2
```